hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanPMigrationStressTest.java

```diff
@@ -0,0 +1,192 @@
+    static final String MAP_NAME = "map";
+
+    private IMap<Integer, Integer> map;
+
+    @BeforeClass
+    public static void setUpClass() {
+        initializeExceptLast(MEMBERS_COUNT, smallInstanceConfig());
+    }
+
+    @Before
+    public void before() {
+        map = instance().getMap(MAP_NAME);
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestPointLookup() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= ITEM_COUNT; i++) {
+            map.put(i, i);
+        }
+        expected.add(new SqlTestSupport.Row(ITEM_TO_LOOKUP, ITEM_TO_LOOKUP));
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.HASH, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1, 500L);
+        mutator.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " WHERE this=" + ITEM_TO_LOOKUP, expected);
+
+        try {
+            mutator.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestSameOrder() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= ITEM_COUNT; i++) {
+            map.put(i, i);
+            expected.add(new SqlTestSupport.Row(ITEM_COUNT - i, ITEM_COUNT - i));
+        }
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.SORTED, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1);
+        mutator.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected);
+
+        try {
+            mutator.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestSameOrderMultipleQueries() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= (ITEM_COUNT / COUNT_DIVIDER); i++) {
+            map.put(i, i);
+            expected.add(new SqlTestSupport.Row((ITEM_COUNT / COUNT_DIVIDER) - i, (ITEM_COUNT / COUNT_DIVIDER) - i));
+        }
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.SORTED, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1, 2500L);
+        QueryThread requester = new QueryThread(expected, instances().length - 1);
+
+        mutator.start();
+        requester.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected);
+
+        try {
+            mutator.join();
+            requester.join();
```

Comment:
I think we should assert here that all queries returned proper results. Asserting in separate thread doesn't fail the test, does it?

Reply:
Enhanced test, thank you!